packages/app-contracts/src/Deploy.tsx

```diff
@@ -0,0 +1,290 @@
+// Copyright 2017-2019 @polkadot/app-contracts authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { SubmittableResult } from '@polkadot/api/SubmittableExtrinsic';
+import { I18nProps } from '@polkadot/ui-app/types';
+
+import BN from 'bn.js';
+import React from 'react';
+import { RouteComponentProps } from 'react-router';
+import { withRouter } from 'react-router-dom';
+
+import { api, withMulti } from '@polkadot/ui-api';
+import keyring from '@polkadot/ui-keyring';
+import { Button, Dropdown, InputBalance, TxButton } from '@polkadot/ui-app';
+import { AccountId, ContractAbi, getTypeDef } from '@polkadot/types';
+import createValues from '@polkadot/ui-params/values';
+
+import ContractModal, { ContractModalProps, ContractModalState } from './Modal';
+import Params from './Params';
+import store from './store';
+import translate from './translate';
+
+type ConstructOptions = Array<{key: string, text: string, value: string}>;
+
+type Props = ContractModalProps & I18nProps & RouteComponentProps & {
+  codeHash?: string
+};
+
+type State = ContractModalState & {
+  codeHash?: string,
+  constructOptions: ConstructOptions,
+  endowment: BN,
+  isHashValid: boolean,
+  params: Array<any>
+};
+
+class Deploy extends ContractModal<Props, State> {
+  constructor (props: Props) {
+    super(props);
+    this.defaultState = {
+      ...this.defaultState,
+      constructOptions: [],
+      endowment: new BN(0),
+      isHashValid: false,
+      params: [],
+      ...this.getCodeState(props.codeHash)
+    };
+    this.state = this.defaultState;
+  }
+
+  componentWillReceiveProps (nextProps: Props, nextState: State) {
+    super.componentWillReceiveProps(nextProps, nextState);
+
+    if (nextProps.codeHash && nextProps.codeHash !== this.props.codeHash) {
+      this.setState(
+        this.getCodeState(nextProps.codeHash)
+      );
+    }
+
+  }
+
+  isContract = true;
```

Comment:
Move to start of class (things do get lost/not seen when somewhere between the methods)